src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -0,0 +1,478 @@
+        replay(response);
+
+        // Act
+        Response createResponse = usersFacade.createOrUpdateUserSettings(request, response, payload.toString());
+
+        // Assert
+        Assertions.assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), createResponse.getStatus());
+    }
+
+    /**
+     * Checks that authenticated requests to update other users of each role by a user with the Student role are rejected.
+     *
+     * @throws Exception - not expected.
+     */
+    @ParameterizedTest
+    @MethodSource("allTestUsersProvider")
+    public void createOrUpdateEndpoint_updateAnotherUserWhileLoggedInAsStudent_failsWithForbidden(RegisteredUserDTO targetUser) throws Exception {
+        // Arrange
+        // log in as an existing student
+        LoginResult studentLogin = loginAs(httpSession, ITConstants.ALICE_STUDENT_EMAIL, ITConstants.ALICE_STUDENT_PASSWORD);
+        HttpServletRequest request = createRequestWithCookies(new Cookie[]{studentLogin.cookie});
+        replay(request);
+
+        // build update request
+        JSONObject payload = new JSONObject()
+                .put("registeredUser", new JSONObject()
+                        .put("email", targetUser.getEmail())
+                        .put("emailVerificationStatus", targetUser.getEmailVerificationStatus())
+                        .put("familyName", ">:)") // this is the update
+                        .put("firstLogin", targetUser.isFirstLogin())
+                        .put("gender", targetUser.getGender())
+                        .put("givenName", targetUser.getGivenName())
+                        .put("id", targetUser.getId())
+                        .put("lastSeen", "1686558490205")
+                        .put("lastUpdated", "1686558490205")
+                        .put("loggedIn", true)
+                        .put("password", JSONObject.NULL)
+                        .put("registeredContext", Set.of())
+                        .put("registrationDate", "1564660299981")
+                        .put("role", targetUser.getRole())
+                        .put("schoolId", targetUser.getSchoolId())
+                )
+                .put("passwordCurrent", JSONObject.NULL);
+
+        HttpServletResponse response = createNiceMock(HttpServletResponse.class);
+        replay(response);
+
+        // Act
+        Response createResponse = usersFacade.createOrUpdateUserSettings(request, response, payload.toString());
+
+        // Assert
+        Assertions.assertEquals(Response.Status.FORBIDDEN.getStatusCode(), createResponse.getStatus());
+    }
+
+    /**
+     * Checks that authenticated requests to update other users of each role by a user with the Teacher role are rejected.
+     *
+     * @throws Exception - not expected.
+     */
+    @ParameterizedTest
+    @MethodSource("allTestUsersProvider")
+    public void createOrUpdateEndpoint_updateAnotherUserWhileLoggedInAsTeacher_failsWithForbidden(RegisteredUserDTO targetUser) throws Exception {
```

Comment:
Adding a test where a member of Staff successfully updates a user's details would be good for fuller endpoint coverage.

Reply:
Done!